src/Symfony/Component/Cache/Traits/FilesystemTrait.php

```diff
@@ -23,6 +23,26 @@
     use FilesystemCommonTrait;
 
     /**
+     * @return bool
+     */
+    public function prune()
+    {
+        $time = time();
+        $pruned = true;
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if ($h = @fopen($file, 'rb')) {
+                if ($time >= (int) $expiresAt = fgets($h)) {
+                    fclose($h);
```

Comment:
the fclose should be outside the if block - at least the file needs to be closed in the "else" path